app/code/Magento/InventoryCatalog/Setup/Operation/MigrateSingleStockData.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Setup\Operation;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\InventoryCatalog\Api\DefaultSourceProviderInterface;
+
+/**
+ * Migrate Single Stock Item Data from CatalogInventory to Inventory Source Item with default source ID
+ */
+class MigrateSingleStockData
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var DefaultSourceProviderInterface
+     */
+    private $defaultSourceProvider;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     * @param DefaultSourceProviderInterface $defaultSourceProvider
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        DefaultSourceProviderInterface $defaultSourceProvider
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->defaultSourceProvider = $defaultSourceProvider;
+    }
+
+    /**
+     * Insert Stock Item to Inventory Source Item by raw MySQL query
+     *
+     * @return void
+     */
+    public function execute()
+    {
+        $defaultSourceId = $this->defaultSourceProvider->getId();
+        $sql = "INSERT INTO inventory_source_item (source_id, sku, quantity, status) 
```

Comment:
Need to add dependency on `Inventory` in module sequence configuration in `module.xml`
Because we operate `inventory_source_item` table which is created in Inventory module

So we need to guarantee that install scheme from `Inventory` is finished before `InventoryCatalog`  installation running

Reply:
Looks like we have double saving of `SourceItem`
For example, I feel both fields in UI (old qty field and new in our grid) with different qty values

So we have save in:
1) In this plugin 
2) In \Magento\InventoryCatalog\Observer\ProcessSourceItemsObserver::execute

And as result, we will have in old system(legacy stockItem) one value, and in new system(source item) different value

Maybe need to improve ProcessSourceItemsObserver (need to work only for multi stock system)
But looks like it will be better to do this in separate task
@maghamed